File: app/src/main/java/android/montreal/devfest/stuff/usecases/StuffUseCaseModule.java
package android.montreal.devfest.stuff.usecases;

import android.montreal.devfest.stuff.policies.RandomStuffVerifier;

import dagger.Module;
import dagger.Provides;

@Module
public class StuffUseCaseModule {

    @Provides
    public ManageRandomStuffUseCase manageRandomStuffUseCase(RandomStuffGetter getter, RandomStuffSetter setter, RandomStuffVerifier verifier) {
        return new ManageRandomStuffUseCaseImpl(verifier, setter, getter);
    }
}

File: stuff_usecases/src/main/java/android/montreal/devfest/stuff/usecases/StuffUseCaseModule.kt
package android.montreal.devfest.stuff.usecases

import android.montreal.devfest.stuff.policies.RandomStuffVerifier

import dagger.Module
import dagger.Provides

@Module
class StuffUseCaseModule {

    @Provides
    fun manageRandomStuffUseCase(getter: RandomStuffGetter, setter: RandomStuffSetter, verifier: RandomStuffVerifier): ManageRandomStuffUseCase {
        return ManageRandomStuffUseCaseImpl(verifier, setter, getter)
    }
}
